backend/capellacollab/settings/configuration/models.py

```diff
@@ -37,6 +38,62 @@ class MetadataConfiguration(core_pydantic.BaseModelStrict):
     environment: str = pydantic.Field(default="-", description="general")
 
 
+class BuiltInLinkItem(str, enum.Enum):
+    GRAFANA = "grafana"
+    PROMETHEUS = "prometheus"
+    DOCUMENTATION = "documentation"
+
+
+class UserRole(str, enum.Enum):
+    USER = "user"
+    ADMIN = "administrator"
+
+
+class BuiltInNavbarLink(core_pydantic.BaseModelStrict):
+    name: str
+    service: BuiltInLinkItem = pydantic.Field(
+        description="Built-in service to link to.",
+    )
+    role: UserRole = pydantic.Field(
+        description="Role required to see this link.",
+    )
+
+
+class CustomNavbarLink(core_pydantic.BaseModelStrict):
+    name: str
+    href: str = pydantic.Field(
+        description="URL to link to.",
+    )
+    role: UserRole = pydantic.Field(
+        description="Role required to see this link.",
+    )
```

Comment:
This is a good example where inheritance is useful. `name` and `role` exist in both NavbarLink classes, so let's add it to the parent class:

```suggestion
class NavbarLink(core_pydantic.BaseModelStrict):
    name: str
    role: UserRole


class BuiltInNavbarLink(NavbarLink):
    service: BuiltInLinkItem = pydantic.Field(
        description="Built-in service to link to.",
    )


class CustomNavbarLink(NavbarLink):
    href: str = pydantic.Field(
        description="URL to link to.",
    )
```